arangod/Agency/FailedFollower.cpp

```diff
@@ -259,7 +276,22 @@ bool FailedFollower::start(bool& aborts) {
           VPackObjectBuilder stillExists(&job);
           job.add("old", VPackValue("FAILED"));
         }
+        // Plan still as we see it:
         addPreconditionUnchanged(job, planPath, planned);
+        // Check that failoverCandidates are still as we inspected them:
+        doForAllShards(_snapshot, _database, shardsLikeMe,
+            [this, &job](Slice plan, Slice current,
+                             std::string& planPath,
+                             std::string& curPath) {
+              // take off "servers" from curPath and add
+              // "failoverCandidates":
+              std::string foCandsPath = curPath.substr(0, curPath.size() - 7);
+              foCandsPath += StaticStrings::FailoverCandidates;
+              auto foCands = this->_snapshot.hasAsSlice(foCandsPath);
+              if (foCands.second) {
+                addPreconditionUnchanged(job, foCandsPath, foCands.first);
```

Comment:
Is "failoverCandidates" a mandatory attribute?
If not, and if it doesn't yet exist, we may end up with no precondition here. In which case we may overwrite changes to the same key by some other concurrent operation?
Maybe this is not possible in reality, but it should be checked.

Reply:
It should be asserted, that `foCands.second` is true. This attribute is mandatory afaik.